cli_tools/diagnostics/main.go

```diff
@@ -0,0 +1,172 @@
+//  Copyright 2018 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//	  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package main
+
+import (
+	"archive/zip"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+var (
+	tmpFolder   = ""
+	errNonFatal = errors.New("method succeeded with errors")
+)
+
+type runner interface {
+	run() (string, error)
+}
+
+type logFolder struct {
+	name  string
+	files []string
+}
+
+func zipFiles(logs []logFolder, outputPath string) error {
+	newFile, err := os.Create(outputPath)
+	if err != nil {
+		return err
+	}
+	defer newFile.Close()
+	writer := zip.NewWriter(newFile)
+	defer writer.Close()
+
+	err = nil
+	for _, folder := range logs {
+		for _, path := range folder.files {
+			file, err := os.Open(path)
+			if err != nil {
+				log.Printf("Error opening file %s for zipping with error %v\n", path, err)
+				err = errNonFatal
+				continue
+			}
+			defer file.Close()
+
+			p := fmt.Sprintf("%s/%s", folder.name, filepath.Base(path))
+			zf, err := writer.Create(p)
+			if err != nil {
+				log.Printf("Error saving file %s to zip with error %v\n", path, err)
+				err = errNonFatal
+				continue
+			}
+
+			if _, err = io.Copy(zf, file); err != nil {
+				log.Printf("Error saving contents of file %s with error %v\n", path, err)
+				err = errNonFatal
+			}
+		}
+	}
+	return err
```

Comment:
Reason for this err return? I dont see you setting it at all. Why not just return nil?

Reply:
Cleaned up the namespace.